pkg/plugins/runtime/k8s/controllers/gatewayapi/attachment/attachment.go

```diff
@@ -66,43 +66,30 @@ func findRouteListenerAttachment(
 			}
 
 			if !selector.Matches(kube_labels.Set(routeNs.GetLabels())) {
-				listeners[l.Name] = NotPermitted
+				listenerAttachments[l.Name] = NotPermitted
 			} else {
-				listeners[l.Name] = Allowed
+				listenerAttachments[l.Name] = Allowed
 			}
 		case gatewayapi_v1.NamespacesFromSame:
-			if gateway.Namespace != routeNs.GetName() {
-				listeners[l.Name] = NotPermitted
+			if gatewayNs != routeNs.GetName() {
+				listenerAttachments[l.Name] = NotPermitted
 			} else {
-				listeners[l.Name] = Allowed
+				listenerAttachments[l.Name] = Allowed
 			}
 		case gatewayapi_v1.NamespacesFromAll:
-			listeners[l.Name] = Allowed
-		}
-	}
-
-	sectionName := ""
-	if refSectionName != nil {
-		sectionName = string(*refSectionName)
-	}
-
-	// Look through the potential Listeners:
-	// If it's our specific listener, we return that status
-	for name, status := range listeners {
-		if string(name) == sectionName {
-			return status, nil
+			listenerAttachments[l.Name] = Allowed
 		}
 	}
 
 	// If we don't find our listener, our ref has NoMatchingParent
-	if sectionName != "" {
+	if len(attemptedAttachments) == 0 {
```

Comment:
is this correct? if it's correct then we could just do this right at the beginning of the function... shouldn't we check `listenerAttachments`?

Reply:
it's correct but yes we can definitely do it at the beginning.